CHANGELOG.md

```diff
@@ -0,0 +1,5 @@
+## 0.1.0 (2024-03-10)
```

Comment:
We already have a changelog file in the docs folder.
https://github.com/kausalflow/hamilflow/blob/main/docs/changelog.md

Not sure which is better. Choose whichever you prefer. 
We need a changelog file in the docs folder. But if you choose this current file, I can easily add a hook in the pre commit confit to automatically copy it to docs. So either ways.